ansible/roles/test/files/ptftests/mtu_test.py

```diff
@@ -0,0 +1,148 @@
+
+'''
+
+#---------------------------------------------------------------------
+# Global imports
+#---------------------------------------------------------------------
+import logging
+import ptf
+import ptf.packet as scapy
+import ptf.dataplane as dataplane
+
+from ptf import config
+from ptf.base_tests import BaseTest
+from ptf.mask import Mask
+from ptf.testutils import *
+
+
+class MtuTest(BaseTest):
+    '''
+    @summary: Overview of functionality
+    Test sends a jumbo ICMP echo frame with MAX MTU size and expects the reply
+    back. It also sends a jumbo frame to a route destination for verifying the 
+    forwarding functionality
+    
+    For the device configured with IP-MTU=9100, PHY-MTU=9122, 
+     - ICMP frame, the packet-len is 9114 (Subtracting 8 bytes of ICMP header from 9122)
+     - IP frame, the packet-len is 9122 (This includes the Layer 2 Ethernet header + FCS)
+    '''
+
+    #---------------------------------------------------------------------
+    # Class variables
+    #---------------------------------------------------------------------
+    DEFAULT_PACKET_LEN = 9122
+    ICMP_HDR_LEN = 8
+
+    def __init__(self):
+        '''
+        @summary: constructor
+        '''
+        BaseTest.__init__(self)
+        self.test_params = test_params_get()
+
+    #---------------------------------------------------------------------
+
+    def setUp(self):
+        self.dataplane = ptf.dataplane_instance
+        self.router_mac = self.test_params['router_mac']
+    
+    #---------------------------------------------------------------------
+
+    def check_icmp_mtu(self):
+        '''
+        @summary: Check ICMP/Ping works for MAX MTU.
+        '''
+        ip_src = "10.0.0.1"
+        ip_dst = "10.0.0.0"
+        src_mac = self.dataplane.get_mac(0, 0)
+
+        pktlen = (self.DEFAULT_PACKET_LEN - self.ICMP_HDR_LEN)
+
+        pkt = simple_icmp_packet(pktlen=pktlen,
+                            eth_dst=self.router_mac,
+                            eth_src=src_mac,
+                            ip_src=ip_src,
+                            ip_dst=ip_dst,
+                            ip_ttl=64)
+
+        exp_pkt = simple_icmp_packet(pktlen=pktlen,
+                            eth_src=self.router_mac,
+                            ip_src=ip_dst,
+                            ip_dst=ip_src,
+                            ip_ttl=64,
+                            icmp_type=0)
+
+        masked_exp_pkt = Mask(exp_pkt)
+        masked_exp_pkt.set_do_not_care_scapy(scapy.Ether, "dst")
+        masked_exp_pkt.set_do_not_care_scapy(scapy.IP, "id")
+        masked_exp_pkt.set_do_not_care_scapy(scapy.IP, "chksum")
+        masked_exp_pkt.set_do_not_care_scapy(scapy.ICMP, "chksum")
```

Comment:
maybe we should match the ttl as well, it is not guaranteed that the icmp reply also use ttl=64. You do not want that factor to fail your test.

Reply:
Yes. Will add "ttl" to _do_not_care field in the reply packet